src/LinuxTracing/SwitchesStatesNamesVisitor.cpp

```diff
@@ -181,14 +185,100 @@ void SwitchesStatesNamesVisitor::Visit(uint64_t event_timestamp,
   std::optional<ThreadStateSlice> state_slice = state_manager_.OnSchedWakeup(
       event_timestamp, event_data.woken_tid, event_data.was_unblocked_by_tid,
       event_data.was_unblocked_by_pid);
+
   if (state_slice.has_value()) {
+    state_slice->set_callstack_id(orbit_base::kInvalidCallstackId);
     listener_->OnThreadStateSlice(std::move(state_slice.value()));
     if (thread_state_counter_ != nullptr) {
       ++(*thread_state_counter_);
     }
   }
 }
 
+void SwitchesStatesNamesVisitor::Visit(uint64_t event_timestamp,
+                                       const SchedWakeupWithStackPerfEventData& event_data) {
+  if (!TidMatchesPidFilter(event_data.woken_tid)) {
+    return;
+  }
+
+  std::optional<ThreadStateSlice> state_slice = state_manager_.OnSchedWakeup(
+      event_timestamp, event_data.woken_tid, event_data.was_unblocked_by_tid,
+      event_data.was_unblocked_by_pid);
+
+  if (state_slice.has_value()) {
+    state_slice->set_callstack_id(orbit_base::kWaitOnCallstack);
+    listener_->OnThreadStateSlice(std::move(state_slice.value()));
+    if (thread_state_counter_ != nullptr) {
+      ++(*thread_state_counter_);
+    }
+  }
+}
+
+void SwitchesStatesNamesVisitor::Visit(uint64_t event_timestamp,
```

Comment:
Nit: This method should be moved up to the non-stack version.

Reply:
Done.